common/rdm/DiscoveryAgent.cpp

```diff
@@ -217,7 +217,9 @@ void DiscoveryAgent::BranchComplete(const uint8_t *data, unsigned int length) {
   OLA_INFO << "BranchComplete, got " << length;
   if (length == 0) {
     // timeout
-    FreeCurrentRange();
+    if (!m_uid_ranges.empty()) {
+      FreeCurrentRange();
```

Comment:
Are we better off fixing this within FreeCurrentRange, in case it gets called from elsewhere?